src/ccl_background.c

```diff
@@ -1014,6 +1014,69 @@ void ccl_comoving_angular_distances(ccl_cosmology * cosmo, int na, double a[],
   }
 }
 
+
+double ccl_angular_diameter_distance(ccl_cosmology * cosmo, double a1, double a2, int* status)
+//Implements Eq. (19) of Hogg, astro-ph/9905116.
+{
+  if(a1>1. || a2>1.) {
+    *status = CCL_ERROR_COMPUTECHI;
+    ccl_cosmology_set_status_message(cosmo,"ccl_background.c: ccl_angular_diameter_distance(): scale factor cannot be larger than 1.");
+    return NAN;
+  } else {
+    if(cosmo->params.Omega_k<0.){
```

Comment:
Is this still true?

Reply:
Which line?

Reply:
That omega_k has to be >=0. Maybe we use what @damonge wrote below?